### ISSUE TITLE
LDAP groups imported with wrong name

### ISSUE THREAD

**Issue by hvancamp** [goharbor/harbor#10437]
**Description:**
It looks like LDAP groups are automatically imported when an LDAP user logins for the first time on Harbor?

Unfortunately with an incorrect name for the LDAP group. Its importing the group name with dn value instead of cn. Even-though in the authentication configuration the "LDAP Group GID" is set to "cn".

No issues when adding an LDAP group to a project via API (api/projects/<project>/members) with "ldap_group_dn" parameter then the group is imported correctly using cn value for the group name.

Example:
Having the following LDAP user to login on Harbor.
uid=myuser,ou=harbor,dc=mydomain,dc=com
memberOf cn=mygroup,ou=harbor,dc=mydomain,dc=com

**Expected behaviour:**
When an LDAP user logins on Harbor, it gets imported in the Harbor database with all its group memberships with CN as name. The Administration/Groups page in Harbor displays:

| Name | Type | DN |
|------------|-------|-----|
|mygroup|LDAP|cn=mygroup,ou=harbor,dc=mydomain,dc=com|

**Actual behaviour:**
When an LDAP user logins on Harbor, it gets imported in the Harbor database with all its group memberships with DN as name. The Administration/Groups page in Harbor displays

| Name | Type | DN |
|------------|-------|-----|
|cn=mygroup,ou=harbor,dc=mydomain,dc=com|LDAP|cn=mygroup,ou=harbor,dc=mydomain,dc=com|

**Steps to reproduce the problem:**
1. Configure Harbor for LDAP authentication with the following LDAP group settings:

|Parameter|Value|
|-|-|
|LDAP Group Base DN|dc=mydomain,dc=com|
|LDAP Group Filter|objectclass=groupOfUniqueNames|
|LDAP Group GID|cn|
|LDAP Group Membership|memberof|
|LDAP Group Scope|Subtree|

2. Login with LDAP user via web page
3. Go to Administration/Groups and it will display the DN value as name for the LDAP user group.

**Versions:**
- harbor version: 1.9.3
- docker engine version: 19.03.1
- docker-compose version: 1.21.2

**Comment by MnrGreg** [comment]
Experiencing the same issue. Groups are imported with the name field matching `DN` and not `cn`.

It also seems that the `ldap_group_search_filter` is being ignored.

**Versions:**
- Harbor version: v1.11.0-e2bab855 (harbor-core:dev)
- LDAP Provider: Active Directory

```json
{
    "auth_mode": "ldap_auth",
    "ldap_url": "ldaps://ldaprck.redacted.com",
    "ldap_base_dn": "dc=redacted,dc=com",
    "ldap_filter": "(&(objectClass=organizationalPerson)(sAMAccountName=*)(!(userAccountControl=514)))",
    "ldap_scope": "2",
    "ldap_search_dn": "CN=redacted,OU=Resources,DC=redacted,DC=com",
    "ldap_search_password": "redacted",
    "ldap_uid": "sAMAccountName",
    "ldap_verify_cert": "False",
    "ldap_group_admin_dn": "CN=Admins,OU=Resources,DC=redacted,DC=com",
    "ldap_group_attribute_name": "cn",
    "ldap_group_base_dn": "OU=Resources,DC=redacted,DC=com",
    "ldap_group_search_filter": "(&(objectClass=group)(cn=CONTAINER*))",
    "ldap_group_search_scope": "1",
    "ldap_group_membership_attribute": "memberOf",
}
```

**Comment by rescuer77** [comment]
Harbor: v1.10.0
LDAP: FreeIPA, version: 4.5.0
ldap_group_search_filter - ignored or not working in harbor
(&(ObjectClass=groupOfNames)(cn=regs*))
However, everything works with ldapsearch command.

**Comment by atoom** [comment]
Same behaviour in 2.0.0-RC1.

My expectation is that only user-groups that match results from the configured group search is added. Currently all user-groups that are in the `LDAP Group Membership` attribute will be imported even though they are irrelevant from a Harbor perspective. As a result the `Add Group Member` ui is very cluttered with many unnecessary groups.

**Comment by hvancamp** [comment]
Still having this issue today on Harbor v1.10.x

**Comment by MnrGreg** [comment]
Same behavior in 2.0.1

It seems Harbor is looking up all the groups to which an authenticating user belongs and createing all of them in the database. When doing so, it ignores the `ldap_group_base_dn`; the `ldap_group_search_filter`; and sets the group names to that of the DN attribute and not whats specified in `ldap_group_attribute_name`.

```yaml
  "ldap_group_attribute_name": "cn",
  "ldap_group_base_dn": "OU=Containers,OU=Resources,DC=redacted,DC=com",
  "ldap_group_search_filter": "(&(objectclass=group)(cn=APR_CONTAINERS*))",
```

```shell
registry=# select * from user_group;
 id  |                                      group_name                                       | group_type |                                     ldap_group_dn                                     |       creation_time        |        update_time         
-----+---------------------------------------------------------------------------------------+------------+---------------------------------------------------------------------------------------+----------------------------+----------------------------
 540 | CN=APR_CyberArk_TSHypervisorServices,OU=CyberArk,OU=Resources,DC=redacted,DC=com          |          1 | cn=apr_cyberark_tshypervisorservices,ou=cyberark,ou=resources,dc=redacted,dc=com          | 2020-07-28 19:11:25.723819 | 2020-07-28 19:11:25.723819
 541 | CN=APR_CyberArk_TSHypervisorServicesApproval,OU=CyberArk,OU=Resources,DC=redacted,DC=com  |          1 | cn=apr_cyberark_tshypervisorservicesapproval,ou=cyberark,ou=resources,dc=redacted,dc=com  | 2020-07-28 19:11:25.725605 | 2020-07-28 19:11:25.725605
```

**Comment by hojerst** [comment]
this seems to be a duplicate to #10940 - I was looking at the source, trying to fix this issue myself and found this ticket, which seems to be scheduled for the 2.1.x release

**Comment by stonezdj** [comment]
It is already fixed in Harbor 2.1
